sos/report/plugins/kubernetes.py

```diff
@@ -121,40 +114,52 @@ def setup(self):
 
         # CNV is not part of the base installation, but can be added
         if self.is_installed('kubevirt-virtctl'):
-            resources.extend(['vms', 'vmis'])
+            self.resources.extend(['vms', 'vmis'])
             self.add_cmd_output('virtctl version')
 
-        for n in knsps:
-            knsp = '--namespace=%s' % n
+        self.collect_per_resource_details()
+        self.collect_all_resources()
+
+    def collect_per_resource_details(self):
+        """ Collect details about each resource in all namespaces """
+        # get all namespaces in use
+        kns = self.collect_cmd_output('%s get namespaces' % self.kube_cmd)
+        # namespace is the 1st word on line, until the line has spaces only
+        kn_output = kns['output'].splitlines()[1:]
+        knsps = [n.split()[0] for n in kn_output if n and len(n.split())]
+
+        for nspace in knsps:
+            knsp = '--namespace=%s' % nspace
+            kube_get_cmd = "get -o json "
```

Comment:
Technically, we can call this line just once before the `for` cycle and save some CPU cycles.